### PR TITLE
Fix linter issues

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @oliver006,

I fixed here some trivial linter issues. PTAL if I should be changing or dropping something.

### CODE REVIEW

File: exporter/redis.go [hunk 1]
```diff
@@ -659,6 +659,8 @@ var errNotFound = errors.New("key not found")
 // getKeyInfo takes a key and returns the type, and the size or length of the value stored at that key.
 func getKeyInfo(c redis.Conn, key string) (info keyInfo, err error) {
 
+	var size int64
```
Comment (oliver006) [inline exporter/redis.go]: What's the reason for pulling this out?

File: exporter/redis.go [hunk 2]
```diff
@@ -711,27 +713,34 @@ func scanForKeys(c redis.Conn, pattern string) ([]string, error) {
...
+			log.Debugln(err)
+		}
 	}
 
 	return keys, nil
 }
 
 // getKeysFromPatterns does a SCAN for a key if the key contains pattern characters
 func getKeysFromPatterns(c redis.Conn, keys []dbKeyPair) (expandedKeys []dbKeyPair, err error) {
+	var keyNames []string
```
Comment (oliver006) [inline exporter/redis.go]: Same, why move this up?

File: exporter/redis.go [hunk 3]
```diff
@@ -747,6 +756,7 @@ func getKeysFromPatterns(c redis.Conn, keys []dbKeyPair) (expandedKeys []dbKeyPa
 }
 
 func (e *Exporter) scrapeRedisHost(scrapes chan<- scrapeResult, addr string, idx int) error {
+	var config []string
```
Comment (oliver006) [inline exporter/redis.go]: Same, why move this up?

File: exporter/redis.go [hunk 4]
```diff
@@ -799,7 +808,8 @@ func (e *Exporter) scrapeRedisHost(scrapes chan<- scrapeResult, addr string, idx
 	isClusterEnabled := strings.Contains(infoAll, "cluster_enabled:1")
 
 	if isClusterEnabled {
-		if clusterInfo, err := redis.String(doRedisCmd(c, "CLUSTER", "INFO")); err == nil {
+		var clusterInfo string
```
Comment (oliver006) [inline exporter/redis.go]: Same, why pull this out?

File: main.go [hunk 5]
```diff
@@ -113,8 +116,14 @@ func main() {
 
 	if *redisMetricsOnly {
 		registry := prometheus.NewRegistry()
-		registry.Register(exp)
-		registry.Register(buildInfo)
+		err = registry.Register(exp)
+		if err != nil {
```
Comment (oliver006) [inline main.go]: --> `if err := registry.Register(exp); err != nil { `

File: main.go [hunk 6]
```diff
@@ -113,8 +116,14 @@ func main() {
...
+			log.Error(err)
+		}
+		err = registry.Register(buildInfo)
+		if err != nil {
```
Comment (oliver006) [inline main.go]: same

File: main.go [hunk 7]
```diff
@@ -98,7 +100,8 @@ func main() {
 	}
 
 	if *scriptPath != "" {
-		script, err := ioutil.ReadFile(*scriptPath)
+		var script []byte
```
Comment (oliver006) [inline main.go]: why move this out?

File: exporter/discovery.go [hunk 8]
```diff
@@ -44,7 +45,10 @@ func LoadRedisFile(fileName string) ([]string, []string, []string, error) {
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	file.Close()
+	err = file.Close()
+	if err != nil {
```
Comment (oliver006) [inline exporter/discovery.go]: `if err := file.Close(); err != nil {`

File: exporter/redis.go (hunk 1 above)
Comment (mjtrangoni) [reply exporter/redis.go]: This is reported by `vetshadow`, a check for shadowed variables. The problem is, as the commit message said, the `err` variable shadowing. 

As I said, if you do not agree, I can drop it.

File: exporter/redis.go (hunk 2 above)
Comment (mjtrangoni) [reply exporter/redis.go]: same here

File: exporter/redis.go (hunk 3 above)
Comment (mjtrangoni) [reply exporter/redis.go]: same here

File: exporter/redis.go (hunk 4 above)
Comment (mjtrangoni) [reply exporter/redis.go]: same here, vetshadow

File: main.go (hunk 7 above)
Comment (mjtrangoni) [reply main.go]: same here, vetshadow

File: exporter/redis.go [hunk 9]
```diff
@@ -711,27 +713,34 @@ func scanForKeys(c redis.Conn, pattern string) ([]string, error) {
 			return keys, fmt.Errorf("invalid response from SCAN for pattern: %s", pattern)
 		}
 
-		k, _ := redis.Strings(arr[1], nil)
+		k, err := redis.Strings(arr[1], nil)
+		if err != nil {
```
Comment (oliver006) [inline exporter/redis.go]: I don't think the is is necessary but if you really want to check for and  error then you should take appropriate action if there is one, e.g not append k to keys if err != nil.

File: exporter/redis.go [hunk 10]
```diff
@@ -711,27 +713,34 @@ func scanForKeys(c redis.Conn, pattern string) ([]string, error) {
...
+			log.Debugln(err)
+		}
 		keys = append(keys, k...)
 
-		if iter, _ = redis.Int(arr[0], nil); iter == 0 {
+		if iter, err = redis.Int(arr[0], nil); iter == 0 {
```
Comment (oliver006) [inline exporter/redis.go]: I think this change is not need nd you should remove 725-727\. you can ignore err here, iter will be zero even if an error is returned and there's no value in logging err, we can simply break.

File: exporter/redis.go (hunk 2 above)
Comment (oliver006) [reply exporter/redis.go]: I'd remove the named return variables, make 736 an instantiation (`:=`) and then the shadowing issue should be resolved.

File: exporter/redis.go (hunk 3 above)
Comment (oliver006) [reply exporter/redis.go]: I'd prefer to not move config up but keep it in the if as is right now, the shadowing isn't a problem here.

File: exporter/redis.go (hunk 4 above)
Comment (oliver006) [reply exporter/redis.go]: same here

File: main.go (hunk 7 above)
Comment (oliver006) [reply main.go]: let's keep this is as is, the shadowing is not going to cause issues.

File: main.go [hunk 11]
```diff
@@ -113,8 +116,12 @@ func main() {
 
 	if *redisMetricsOnly {
 		registry := prometheus.NewRegistry()
-		registry.Register(exp)
-		registry.Register(buildInfo)
+		if err = registry.Register(exp); err != nil {
+			log.Error(err)
```
Comment (oliver006) [inline main.go]: Let's just  `MustRegister` - if an error occurs (which it shouldn't) then continuing doesn't make much sense anyway but we should fail.

File: main.go [hunk 12]
```diff
@@ -113,8 +116,12 @@ func main() {
...
+		}
+		if err = registry.Register(buildInfo); err != nil {
```
Comment (oliver006) [inline main.go]: same

File: main.go [hunk 13]
```diff
@@ -133,6 +140,9 @@ func main() {
 </body>
 </html>
 						`))
+		if err != nil {
```
Comment (oliver006) [inline main.go]: this doesn't really add anything useful, I think it's ok to ignore err here (and remove it from line 127).

**Comment by mjtrangoni** [comment]
@oliver006 PTAL. I dropped the vet shadow commit.

**Comment by oliver006** [comment]
@mjtrangoni - thanks for making the changes. There are still a few review comments that were not addressed, esp in main.go and in exporter/redis.go around lines 717 and 722
Once those addressed we should be good to 🚢 this.

**Comment by oliver006** [comment]
Closing - please re-open if this is still being worked on.

**Pull request closed without merging**